docs: add sub-recipes topic (#3241)
block/goose: documentation/docs/guides/recipes/recipe-reference.md

| `prompt` | String | A template prompt that can include parameter substitutions; required in headless (non-interactive) mode |
| `parameters` | Array | List of parameter definitions |
| `extensions` | Array | List of extension configurations |

> | `sub_recipes` | Array | List of sub-recipes |

Reviewer: ```suggestion
| `sub_recipes` | Array | List of subrecipes |
```

    description: "For searching logs using Presidio"
```

## Sub-Recipes

The `sub_recipes` field specifies the [sub-recipes](/docs/guides/recipes/sub-recipes) that the main recipe calls to perform specific tasks. Each sub-recipe in the array has the following structure:

> ### Sub-Recipe Fields

Reviewer: ```suggestion
### SubRecipe Fields
```

| Field | Type | Description |
|-------|------|-------------|
| `name` | String | Unique identifier for the sub-recipe |
| `path` | String | Relative or absolute path to the sub-recipe file |

> | `values` | Object | (Optional) Pre-configured parameter values that are passed to the sub-recipe |

Reviewer: ```suggestion
| `values` | Object | (Optional) Pre-configured parameter values that are passed to the subrecipe |
```
